framework/scripts/generate_test_list.py

```diff
@@ -80,17 +80,37 @@ def main():
         decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
         f.write('\n'.join(decls))
         f.write('\n\n')
+        if len(test_lists) > 0:
+            f.write('struct TestList {\n')
+            f.write('    uint64_t features;\n')
+            f.write('    const char* name;\n')
+            f.write('    const std::span<struct test* const> content;\n')
+            f.write('};\n\n')
+
         for name, tests in test_lists.items():
-            f.write(f'static constexpr struct test * const {name}_test_list[] = {{')
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
             test_list = [ f'    &_test_{test}' for test in tests ]
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
+            f.write(f'static const struct TestList {name}_test_list = {{ {"cpu_" + name if name != "default" else 0}, "{name}", {name}_test_list_ }};\n\n')
+        
+        # write list detector function
+        f.write('optional<const span<struct test * const>> auto_detect() {\n')
+        for name, tests in test_lists.items():
+            f.write(f'    if (cpu_features == {name}_test_list.features) return {name}_test_list.content;\n')
```

Comment:
to check

Reply:
to do it like in detect_cpu_arch(), that is with masking `(cpu_features & arch) == arch`, it would require from user to define the lists in `-Dbuiltin_test_list=default:tests/data/test_list_default.csv,icx:tests/data/test_list_icx.csv...`  in a correct order. This is rather insider knowledge, do we want to require user to know that?

Reply:
for example, core2 testlist should go as the last one